yarn-project/end-to-end/src/guides/dapp_testing.test.ts

```diff
@@ -20,86 +24,103 @@ describe('guides/dapp/testing', () => {
-        await token.methods.transfer(20n, recipient.getAddress()).send().wait();
-        expect(await token.methods.getBalance(recipient.getAddress()).view()).toEqual(20n);
+      it('increases recipient funds on mint', async () => {
+        expect(await token.methods.balance_of_private({ address: recipient.getAddress() }).view()).toEqual(0n);
+        const secret = Fr.random();
+        const secretHash = await computeMessageSecretHash(secret);
+        expect((await token.methods.mint_private(20n, secretHash).send().wait()).status).toEqual(TxStatus.MINED);
+        expect(
+          (await token.methods.redeem_shield({ address: recipient.getAddress() }, 20n, secret).send().wait()).status,
+        ).toEqual(TxStatus.MINED);
+        expect(await token.methods.balance_of_private({ address: recipient.getAddress() }).view()).toEqual(20n);
       });
     });
   });
 
   describe('on local sandbox', () => {
     beforeAll(async () => {
-      const { SANDBOX_URL = 'http://localhost:8080' } = process.env;
       const rpc = createAztecRpcClient(SANDBOX_URL);
       await waitForSandbox(rpc);
     });
 
     // docs:start:sandbox-example
     describe('private token contract', () => {
-      const { SANDBOX_URL = 'http://localhost:8080' } = process.env;
-
       let rpc: AztecRPC;
       let owner: AccountWallet;
       let recipient: AccountWallet;
-      let token: PrivateTokenContract;
+      let token: TokenContract;
 
       beforeEach(async () => {
         rpc = createAztecRpcClient(SANDBOX_URL);
         owner = await createAccount(rpc);
         recipient = await createAccount(rpc);
-        token = await PrivateTokenContract.deploy(owner, 100n, owner.getAddress()).send().deployed();
+        token = await TokenContract.deploy(owner).send().deployed();
+        expect((await token.methods._initialize({ address: owner.getAddress() }).send().wait()).status).toEqual(
+          TxStatus.MINED,
+        );
       }, 30_000);
 
-      it('increases recipient funds on transfer', async () => {
-        expect(await token.methods.getBalance(recipient.getAddress()).view()).toEqual(0n);
-        await token.methods.transfer(20n, recipient.getAddress()).send().wait();
-        expect(await token.methods.getBalance(recipient.getAddress()).view()).toEqual(20n);
+      it('increases recipient funds on mint', async () => {
+        expect(await token.methods.balance_of_private({ address: recipient.getAddress() }).view()).toEqual(0n);
+        const secret = Fr.random();
+        const secretHash = await computeMessageSecretHash(secret);
+        expect((await token.methods.mint_private(20n, secretHash).send().wait()).status).toEqual(TxStatus.MINED);
+        expect(
+          (await token.methods.redeem_shield({ address: recipient.getAddress() }, 20n, secret).send().wait()).status,
+        ).toEqual(TxStatus.MINED);
+        expect(await token.methods.balance_of_private({ address: recipient.getAddress() }).view()).toEqual(20n);
```

Comment:
As this is being used in the example we should probably limit the level of chaining in the calls

Reply:
Ye, just wasted a lot of space otherwise, but maybe fine to include the `waitForSuccess` or just do the wait and ignore the status as we are doing in quite a few places anyway 🤷